pentasharp/Controllers/DashboardController.cs

```diff
@@ -2,16 +2,70 @@
 using System.Diagnostics;
 using WebApplication1.Models;
 using WebApplication1.Filters;
+using AutoMapper;
+using pentasharp.Data;
+using pentasharp.Interfaces;
+using pentasharp.ViewModel.Bus;
+using pentasharp.Models.TaxiRequest;
+using pentasharp.Services;
+using pentasharp.Models.Enums;
+using pentasharp.Models.Utilities;
+using System.Linq;
 
 namespace WebApplication1.Controllers
 {
-    [ServiceFilter(typeof(AdminOnlyFilter))]
+    [ServiceFilter(typeof(AdminBaseFilter))]
+    [Route("Dashboard")]
     public class DashboardController : Controller
     {
+        private readonly ITaxiCompanyService _taxiCompanyService;
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
+
+        public DashboardController(AppDbContext context, IMapper mapper, ITaxiCompanyService taxiCompanyService)
+        {
+            _taxiCompanyService = taxiCompanyService;
+            _context = context;
+            _mapper = mapper;
+        }
 
         public IActionResult Dashboard()
         {
             return View();
         }
+
+        [HttpGet("Bus")]
+        public IActionResult Bus()
+        {
+            var companies = _context.BusCompanies.ToList();
+            var viewModel = new ManageBusCompanyViewModel
```

Comment:
as per other comments, move logic to the service of this endpoint and all others in this controller